src/pipeline_v3/tests/security/test_comprehensive_security.py

```
@@ -0,0 +1,361 @@
+
+        # Whitelisted domains should pass even if private
+        url = "http://trusted.internal/doc.pdf"
+        validated = URLSecurityValidator.validate_url(url, custom_whitelist=whitelist)
+        assert validated == url
+
+        # Subdomains should also work
+        url = "http://sub.api.company.com/doc.pdf"
+        validated = URLSecurityValidator.validate_url(url, custom_whitelist=whitelist)
+        assert validated == url
+
+    @patch('socket.gethostbyname')
```

Ruff (0.11.9)

Single quotes found but double quotes preferred

Replace single quotes with double quotes

(Q000)